controllers/pipelines/runconfiguration_controller_decoupled_test.go

```diff
@@ -369,6 +369,31 @@ var _ = Describe("RunConfiguration controller k8s integration", Serial, func() {
 		})
 	})
 
+	When("RunSpec changes", func() {
+		It("onChange trigger creates a run when run template has changed", func() {
```

Comment:
Wish we had unit tests for this part to check that it is not triggering when it shouldn't.